runtime/js/40_testing.js

```diff
@@ -1013,32 +1205,31 @@
       createTestFilter(filter),
     );
 
+    let groups = new Set([undefined]);
+    const benchmarks = ArrayPrototypeFilter(filtered, (bench) => !bench.ignore);
+
+    for (const bench of benchmarks) {
+      bench.group ||= undefined;
+      groups.add(bench.group);
+    }
+
+    groups = ArrayFrom(groups);
+    ArrayPrototypeSort(
+      benchmarks,
+      (a, b) => groups.indexOf(a.group) - groups.indexOf(b.group),
```

Comment:
Is this sorting intended to be stable? I believe it will be unstable because it originated from a `Set`?

Reply:
But an array is created from the set. Don't you need to sort the groups array?

Reply:
Set maintains insertion order